Simplified/NYPLAnnotations.swift

```diff
@@ -12,177 +12,158 @@ final class NYPLAnnotations: NSObject {
   
   class func postLastRead(_ book:NYPLBook, cfi:NSString)
   {
-    //Last Read currently disabled
-//    if (NYPLAccount.shared().hasBarcodeAndPIN())
-//    {
-//      let parameters = [
-//        "@context": "http://www.w3.org/ns/anno.jsonld",
-//        "type": "Annotation",
-//        "motivation": "http://librarysimplified.org/terms/annotation/idling",
-//        "target":[
-//          "source": book.identifier,
-//          "selector": [
-//            "type": "oa:FragmentSelector",
-//            "value": cfi
-//          ]
-//        ]
-//        ] as [String : Any]
-//      
-//      let url = NYPLConfiguration.mainFeedURL().appendingPathComponent("annotations/")
-//      
-//      postJSONRequest(url, parameters, NYPLAnnotations.headers)
-//    }
+    if (NYPLAccount.shared().hasBarcodeAndPIN())
+    {
+      let parameters = [
+        "@context": "http://www.w3.org/ns/anno.jsonld",
+        "type": "Annotation",
+        "motivation": "http://librarysimplified.org/terms/annotation/idling",
+        "target":[
+          "source":  book.identifier,
+          "selector": [
+            "type": "oa:FragmentSelector",
+            "value": cfi
+          ]
+        ],
+        "body": [
+          "http://librarysimplified.org/terms/time" : iso8601CurrentTime(),
+          "http://librarysimplified.org/terms/device" : NYPLAccount.shared().deviceID
+        ]
+        ] as [String : Any]
+      
+      let url = NYPLConfiguration.mainFeedURL().appendingPathComponent("annotations/")
+      
+      postJSONRequest(book, url, parameters, NYPLAnnotations.headers)
+    }
   }
   
   class func sync(_ book:NYPLBook, completionHandler: @escaping (_ responseObject: String?, _ error: NSError?) -> ()) {
-    //Sync Currently Disabled
-//    syncLastRead(book, completionHandler: completionHandler)
+    syncLastRead(book, completionHandler: completionHandler)
   }
   
-  private class func postJSONRequest(_ url: URL, _ parameters: [String:Any], _ headers: [String:String]?)
+  private class func postJSONRequest(_ book: NYPLBook, _ url: URL, _ parameters: [String:Any], _ headers: [String:String]?)
   {
-
-    //FIXME: this network code needs to be tested before being re-implemented. Alamofire was removed. Remove this comment after turning annotations/sync back on.
+    let jsonData: Data?
+    do {
+      jsonData = try JSONSerialization.data(withJSONObject: parameters, options: [.prettyPrinted])
+    } catch {
+      Log.error(#file, "Network request abandoned. Could not create JSON from given parameters.")
+      return
+    }
```

Comment:
Not quite. I'm saying that due to the way you've written in, the type system will not understand `jsonData` is non-nil because you've declared it as type `Data?`. If you use the `guard let` form instead with `try?` to convert exceptions to `nil`, you can assign `jsonData` the non-nil type `Data`. It happens not to matter in this case because all uses of `jsonData` accept `nil`, but it might make the code a bit clearer.